platform-sdk/swirlds-common/src/test/java/com/swirlds/common/io/utility/RecycleBinTests.java

```diff
@@ -204,8 +205,71 @@ void recycleDuplicateFileTest() throws IOException {
         recycleBin.recycle(path);
         assertFalse(Files.exists(path));
         validateFile(recycledPath, "baz");
+    }
 
-        recycleBin.clear();
-        assertFalse(Files.exists(recycledPath));
+    @Test
+    @DisplayName("Files Deleted After Time Passes")
+    void filesDeletedAfterTimePasses() throws IOException, InterruptedException {
+        final FakeTime time = new FakeTime(Instant.now(), Duration.ZERO);
+
+        final Configuration customConfiguration = new TestConfigBuilder()
+                .withValue("state.savedStateDirectory", testDirectory.toString())
+                .withValue("recycleBin.collectionPeriod", "1ns")
+                .getOrCreateConfig();
+
+        final Duration maximumFileAge =
+                customConfiguration.getConfigData(RecycleBinConfig.class).maximumFileAge();
+
+        final RecycleBinImpl recycleBin = new RecycleBinImpl(
+                customConfiguration, new NoOpMetrics(), getStaticThreadManager(), time, new NodeId(0));
+        recycleBin.start();
+
+        // Recycle some files.
+        final Path path1 = testDirectory.resolve("file1.txt");
+        writeFile(path1, "file1");
+
+        final Path path2 = testDirectory.resolve("file2.txt");
+        writeFile(path2, "file2");
+
+        final Path path3 = testDirectory.resolve("file3.txt");
+        writeFile(path3, "file3");
+
+        recycleBin.recycle(path1);
+        recycleBin.recycle(path2);
+        recycleBin.recycle(path3);
```

Comment:
I suggest also making sure cleanup works with a recycles directory, in addition to plain files

Reply:
test added